### PR TITLE
Feat/auth UI

### PR DESCRIPTION
**Pull request by Jhoscar22**
* **Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes/features)
- [x] Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
SignIn UI, reusable NavBar component (without menu), reusable TextInput and InputField components. Small tweak in IbmButton component.

* **What is the current behavior?** (You can also link to an open issue here)
No SignIn route.

* **What is the new behavior (if this is a feature change)?**
New route at /signin

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
No.

* **Other information**:

### CODE REVIEW

**Comment by Jhoscar22** [comment]
Added SignIn UI and made some reusable components for Forms

File: data-dashboard/pages/signin.tsx [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+  marginRight: '202px',
+};
+
+function SignIn() {
+  return (
+    <div>
+      <NavBar />
+      <Body>
+        <Content>
+          <SectionHeader>Iniciar sesión</SectionHeader>
```
Comment (antonieto) [inline data-dashboard/pages/signin.tsx:85]: We need to move this to a layout component and use [NextJS layouts](https://nextjs.org/docs/basic-features/layouts)

Comment (antonieto) [reply data-dashboard/pages/signin.tsx:85]: But we can do that later :)

File: data-dashboard/.eslintrc.json [hunk 2]
```diff
@@ -17,6 +17,16 @@
   },
   "rules": {
     "camelcase": "error",
+    "max-len": [
```
Comment (antonieto) [inline data-dashboard/.eslintrc.json:20]: Nice

File: data-dashboard/lib/components/TextInput/TextInput.tsx [hunk 3]
```diff
@@ -0,0 +1,55 @@
+import styled from 'styled-components';
+import React from 'react';
+
+export interface TextInputProps {
```
Comment (antonieto) [inline data-dashboard/lib/components/TextInput/TextInput.tsx:4]: Exporting the props of a component is a sign of tight coupling between components.
```suggestion
interface TextInputProps {
```

File: data-dashboard/lib/components/InputField/InputField.tsx [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+// components yet, please add them here if you create them.
+
+import styled from 'styled-components';
+import TextInput, { TextInputProps } from '../TextInput/TextInput';
+
+interface InputFieldProps {
+  label: string;
+  name?: string;
+  inputChild: typeof TextInput;
+  inputProps: TextInputProps;
```
Comment (antonieto) [inline data-dashboard/lib/components/InputField/InputField.tsx:13]: I am not a huge fan of this pattern but this line makes it possible to skip type `TextInputProps`
```suggestion
  inputProps: React.ComponentProps<typeof TextInputProps>;
```

**Review by antonieto — APPROVED** [review]
good so far!

File: data-dashboard/lib/components/InputField/InputField.tsx (hunk 4 above)
Comment (Jhoscar22) [reply data-dashboard/lib/components/InputField/InputField.tsx:13]: I will surely add this later with other changes.

File: data-dashboard/lib/components/TextInput/TextInput.tsx (hunk 3 above)
Comment (Jhoscar22) [reply data-dashboard/lib/components/TextInput/TextInput.tsx:4]: I don't get why this would be bad, I will make some research.

**Review by DavBot02 — APPROVED** [review]

**Review by antonieto — APPROVED** [review]

File: data-dashboard/pages/signin.tsx [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+          <Description>Ingresa tus datos para iniciar sesión</Description>
+          <Form>
+            <InputFields>
+              <InputField
```
Comment (Int-Angel) [inline data-dashboard/pages/signin.tsx:89]: Maybe we can include another export inside InputField.tsx that already exports an InputField with an inputChild TexInput, I think we are going to use that a lot

**Review by Int-Angel — APPROVED** [review]
LGTM!

**Review by paotg123 — COMMENTED** [review]
👍🏼

**Pull request merged**